common/djangoapps/third_party_auth/models.py

```diff
@@ -109,6 +118,12 @@ class Meta(object):
         app_label = "third_party_auth"
         abstract = True
 
+    def clean(self):
+        """ Ensure that either `icon_class` or `icon_image` is set """
+        super(ProviderConfig, self).clean()
+        if bool(self.icon_class) == bool(self.icon_image):
+            raise ValidationError('Either an icon class or an icon image must be given')
```

Comment:
Should say, 'Either an icon class or an icon image must be given (but not both)' because specifying both could also trigger this error.